roborock/api.py

```diff
@@ -103,6 +103,8 @@ def __init__(self, endpoint: str, device_info: DeviceData) -> None:
         self._last_disconnection = self.time_func()
         self.keep_alive = KEEPALIVE
         self._diagnostic_data: dict[str, dict[str, Any]] = {}
+        self.dnd_timer: DnDTimer | None = None
+        self.valley_timer: ValleyElectricityTimer | None = None
```

Comment:
Not sure if this is the best way, but I'm trying to avoid doing the coordinator, and rather than place it in device info, cause then I'd have to make a new function call on top of get_dnd_timer to update it, rather than just setting it here.

Reply:
I thought about something similar. Having an object to hold all executed queries with evicting and caching based on GET / (SET / CHANGE) commands.
You can check them out in roborock_typing CacheableCommands

Reply:
Oh that's nice and probably much better long term than this